src/GitVersionCore/Configuration/ConfigurationBuilder.cs

```diff
@@ -162,6 +163,10 @@ private static void ValidateConfiguration(Config config)
                 {
                     throw new ConfigurationException($"Branch configuration '{name}' is missing required configuration 'source-branches'{System.Environment.NewLine}" + "See https://gitversion.net/docs/configuration/ for more info");
                 }
+
+                var missingSourceBranches = sourceBranches.Where(sb => config.GetConfigForBranch(sb) == null).ToArray();
+                if (missingSourceBranches.Any())
+                    throw new ConfigurationException($"Branch configuration '{name}' defines these 'source-branches' that are not configured: '[{string.Join(",", missingSourceBranches)}]'{System.Environment.NewLine}" + "See https://gitversion.net/docs/configuration/ for more info");
```

Comment:
What would happen here if source branches were configured to a branch that shouldn't require any explicit configuration, such as `master` or `develop`? Can you add a test to ensure it doesn't fail when introducing this check?